app/models/term.rb

```diff
@@ -68,6 +87,22 @@ def calculate_categorizations
 
   private
 
+  # register_fingerprint method gets called before a Term record is saved, ensuring that Terms should always have a
+  # related Fingerprint method.
+  def register_fingerprint
+    new_record = {
+      value: Fingerprint.calculate(phrase)
+    }
+    self.fingerprint = Fingerprint.find_or_create_by(new_record)
```

Comment:
Non blocking. You may intentionally be making the hash, but I wanted to share it is not necessary and you could instead just pass the key and value directly without the intermediary object creation.

`self.fingerprint = Fingerprint.find_or_create_by(value: Fingerprint.calculate(phrase))`